src/pyedb/dotnet/edb_core/padstack.py

```diff
@@ -1648,35 +1648,60 @@ def merge_via(self, contour_boxes, net_filter=None, start_layer=None, stop_layer
             instances = self.get_padstack_instances_id_intersecting_polygon(
                 points=contour_box, padstack_instances_index=instances_index
             )
-            if net_filter:
-                instances = [self.instances[id] for id in instances if not self.instances[id].net.name in net_filter]
-            net = self.instances[instances[0]].net.name
-            instances_pts = np.array([self.instances[id].position for id in instances])
-            convex_hull_contour = ConvexHull(instances_pts)
-            contour_points = list(instances_pts[convex_hull_contour.vertices])
-            layer = list(self._pedb.stackup.layers.values())[0].name
-            polygon = self._pedb.modeler.create_polygon(main_shape=contour_points, layer_name=layer)
-            polygon_data = polygon.polygon_data
-            polygon.delete()
-            new_padstack_def = generate_unique_name(self.instances[instances[0]].definition.name)
-            if not self.create(
-                padstackname=new_padstack_def,
-                pad_shape="Polygon",
-                antipad_shape="Polygon",
-                pad_polygon=polygon_data,
-                antipad_polygon=polygon_data,
-                polygon_hole=polygon_data,
-                start_layer=start_layer,
-                stop_layer=stop_layer,
-            ):
-                self._logger.error(f"Failed to create padstack definition {new_padstack_def}")
-            merged_instance = self.place(position=[0, 0], definition_name=new_padstack_def, net_name=net)
-            merged_via_ids.append(merged_instance.id)
-            [self.instances[id].delete() for id in instances]
+            if instances:
```

Comment:
If possible, please use below code. It will be easier to review. (less line of change). 

Please raise error instead of log error.

if not instance:
    raise ValueError(f"No padstack instances found inside {contour_box}")